internal/services/kusto/kusto_cosmosdb_data_connection_resource.go

```diff
@@ -0,0 +1,248 @@
+
+			existing, err := client.Get(ctx, id)
+			if err != nil && !response.WasNotFound(existing.HttpResponse) {
+				return fmt.Errorf("checking for presence of existing %s: %+v", id, err)
+			}
+			if !response.WasNotFound(existing.HttpResponse) {
+				return metadata.ResourceRequiresImport(r.ResourceType(), id)
+			}
+
+			properties := dataconnections.CosmosDbDataConnectionProperties{
+				CosmosDbAccountResourceId: model.CosmosDbAccountId,
+				CosmosDbContainer:         model.CosmosDbContainer,
+				CosmosDbDatabase:          model.CosmosDbDatabase,
+				TableName:                 model.TableName,
+				ManagedIdentityResourceId: model.ManagedIdentityId,
+			}
+
+			if model.MappingRuleName != "" {
+				properties.MappingRuleName = &model.MappingRuleName
+			}
+
+			if model.RetrievalStartDate != "" {
+				properties.RetrievalStartDate = &model.RetrievalStartDate
+			}
+
+			l := location.Normalize(model.Location)
+
+			dataConnection := dataconnections.CosmosDbDataConnection{
+				Location:   &l,
+				Name:       &model.Name,
+				Properties: &properties,
+			}
+
+			if err := client.CreateOrUpdateThenPoll(ctx, id, dataConnection); err != nil {
+				return fmt.Errorf("creating %s: %+v", id, err)
+			}
+
+			metadata.SetID(id)
+			return nil
+		},
+	}
+}
+
+func (r CosmosDBDataConnectionResource) Read() sdk.ResourceFunc {
+	return sdk.ResourceFunc{
+		Timeout: 5 * time.Minute,
+		Func: func(ctx context.Context, metadata sdk.ResourceMetaData) error {
+			client := metadata.Client.Kusto.DataConnectionsClient
+
+			id, err := dataconnections.ParseDataConnectionID(metadata.ResourceData.Id())
+			if err != nil {
+				return fmt.Errorf("parsing %s: %+v", metadata.ResourceData.Id(), err)
+			}
+
+			resp, err := client.Get(ctx, *id)
+			if err != nil {
+				if response.WasNotFound(resp.HttpResponse) {
+					return metadata.MarkAsGone(id)
+				}
+
+				return fmt.Errorf("retrieving %s: %+v", *id, err)
+			}
+
+			state := CosmosDBDataConnectionModel{
+				Name:              id.DataConnectionName,
+				ResourceGroupName: id.ResourceGroupName,
+				ClusterName:       id.ClusterName,
+				DatabaseName:      id.DatabaseName,
+			}
+
+			if model := resp.Model; model != nil {
+				cosmosDbModel := (*model).(dataconnections.CosmosDbDataConnection)
+				state.Location = location.Normalize(*cosmosDbModel.Location)
+
+				if properties := cosmosDbModel.Properties; properties != nil {
+					state.CosmosDbAccountId = properties.CosmosDbAccountResourceId
+					state.CosmosDbContainer = properties.CosmosDbContainer
+					state.CosmosDbDatabase = properties.CosmosDbDatabase
+					state.TableName = properties.TableName
+					state.ManagedIdentityId = properties.ManagedIdentityResourceId
+
+					if properties.MappingRuleName != nil {
+						state.MappingRuleName = *properties.MappingRuleName
+					}
+
+					if properties.RetrievalStartDate != nil {
+						state.RetrievalStartDate = *properties.RetrievalStartDate
+					}
```

Comment:
Use `pointer.From`, it'll do the nil check and set an empty value if the property isn't present in the response
```suggestion
					state.MappingRuleName = pointer.From(properties.MappingRuleName)
					state.RetrievalStartDate = pointer.From(properties.RetrievalStartDate)
```